File: Python Scripts/Bioinformatics/ntCounter.py
dnaStrand = input("Please enter a DNA strand (no longer than 1000 nt) whose nucleotide count (# of A's, C's, etc.) you would like to know: \n")

valid = 'ACTG'

def nt_loop(dnaStrand):
    nt_A = 0
    nt_C = 0
    nt_G = 0
    nt_T = 0

    for position in range(len(dnaStrand)):
        if dnaStrand[position] == "A":
            nt_A += 1
        elif dnaStrand[position] == "C":
            nt_C += 1
        elif dnaStrand[position] == "G":
            nt_G += 1 
        elif dnaStrand[position] == "T":
            nt_T += 1  
        nt_num = [nt_A, nt_C, nt_G, nt_T]
    return nt_num

for letter in dnaStrand:
    if(len(dnaStrand) <= 1000 and letter in dnaStrand in valid):
        print(*nt_loop(dnaStrand))
else:
    print("Please enter a valid DNA strand (contains A, C, T, G) with less than 1000 nt")

File: C# Code/Bioinformatics/ntCounter.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Linq;
using ntCounter;

namespace ntCounter
{
        class ntCount
        {
                public static void  Main(string[] args)
            {
                Console.WriteLine("Please enter a DNA strand (no longer than 1000 nt) whose nucleotide count you would like to know: ");

                string dnaStrand = Console.ReadLine();
                int d = dnaStrand.Length;

                for(int j = 0; j < d; j++)
                {
                    if(d <= 1000 && (dnaStrand.Contains('A') || dnaStrand.Contains('C') || dnaStrand.Contains('T') || dnaStrand.Contains('G')))
                    {
                        Console.WriteLine("The nucleotide counts for the DNA strand entered are (A, C, G, and T): ");
                        ntCount.ntLoop(dnaStrand);
                    } else {
                Console.WriteLine("Please enter a valid DNA strand (contains A, C, T, G) with less than 1000 nt.");
                    }  
                }
            }
            public static void ntLoop(string dnaStrand)
            {
            int nt_A = 0;
            int nt_C = 0;
            int nt_G = 0;
            int nt_T = 0;
            
            int d = dnaStrand.Length;

            for(int i = 0; i < d; i++)
            {
                if(dnaStrand[i] == 'A')
                {
                    nt_A += 1;
                }
                else if(dnaStrand[i] == 'C')
                {
                    nt_C += 1;
                }
                else if(dnaStrand[i] == 'G')
                {
                    nt_G += 1;
                }
                else if(dnaStrand[i] == 'T')
                {
                    nt_T += 1;
                }
            }
            Console.WriteLine("A: " + nt_A);
            Console.WriteLine("C: " + nt_C);
            Console.WriteLine("G: " + nt_G);
            Console.WriteLine("T: " + nt_T);
            }
        }
    }
   
